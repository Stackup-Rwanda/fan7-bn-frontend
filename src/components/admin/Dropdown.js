import React, { Component } from 'react';
import PropTypes from 'prop-types';
import '../../assets/scss/style.scss';
import more from '../../assets/icons/icons8-collaborator-male-24.png';

class Dropdown extends Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(e) {
    const { updateRole } = this.props;
    updateRole(e.target.dataset.role);
  }

  render() {
    return (
      <div className="Dropdown">
        <div className="Dropdown_main">
          <div className="Dropdown_main_title">
            <h3>Change user role to :</h3>
          </div>
          <div>
            <ul className="Dropdown_main_body">
              <li className="Dropdown_main_body_element1" data-role="manager" onClick={this.handleClick}>
                <img src={more} alt="icon" />
                Manager
              </li>
              <li className="Dropdown_main_body_element2" data-role="host-supplier" onClick={this.handleClick}>
                <img src={more} alt="icon" />
                Host-supplier
              </li>
              <li className="Dropdown_main_body_element3" data-role="travel-administrator" onClick={this.handleClick}>
                <img src={more} alt="icon" />
                Travel Admin
              </li>
              <li className="Dropdown_main_body_element4" data-role="requester" onClick={this.handleClick}>
                <img src={more} alt="icon" />
                Requester
              </li>
            </ul>
          </div>
        </div>
      </div>
    );
  }
}

Dropdown.propTypes = {
  updateRole: PropTypes.func.isRequired,
};

export default Dropdown;
