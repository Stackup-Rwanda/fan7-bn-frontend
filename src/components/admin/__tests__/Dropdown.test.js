import React from 'react';
import { shallow } from 'enzyme';
import Dropdown from '../Dropdown';

describe('Dropdown Component Test', () => {
    it('should return true if component exists', () => {
        const props = {
            loading: false,
            users: [{
                email: "e.kskemc@gmail.com",
                role: "requester"
            }],
        };
        const wrapper = shallow(<Dropdown {...props} />);

        expect(wrapper.find('.Dropdown').exists()).toBe(true);
    });

    it('Should handle click', () => {
        const event = {
            preventDefault: jest.fn(),
            target: {
                dataset: {
                    role: 'manager'
                }
            },
        };
        const props = {
            updateRole: jest.fn(),
            loading: false,
        };
        const wrapper = shallow(<Dropdown {...props} />);

        wrapper.find('.Dropdown_main_body_element1').first().simulate('click', event);

        expect(props.updateRole).toHaveBeenCalled();
    })

});

describe('<Dropdown /> shallow rendering tests', () => {
    const props = {
            loading: false,
            users: [{
                email: "e.kskemc@gmail.com",
                role: "requester"
            }],
    };
    it('matches the snapshot', () => {
        const tree = shallow(<Dropdown {...props} />);
        expect(tree).toMatchSnapshot();
    });

});