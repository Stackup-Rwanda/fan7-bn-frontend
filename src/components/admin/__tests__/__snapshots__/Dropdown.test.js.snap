// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> shallow rendering tests matches the snapshot 1`] = `
<div
  className="Dropdown"
>
  <div
    className="Dropdown_main"
  >
    <div
      className="Dropdown_main_title"
    >
      <h3>
        Change user role to :
      </h3>
    </div>
    <div>
      <ul
        className="Dropdown_main_body"
      >
        <li
          className="Dropdown_main_body_element1"
          data-role="manager"
          onClick={[Function]}
        >
          <img
            alt="icon"
            src={
              Object {
                "process": [Function],
              }
            }
          />
          Manager
        </li>
        <li
          className="Dropdown_main_body_element2"
          data-role="host-supplier"
          onClick={[Function]}
        >
          <img
            alt="icon"
            src={
              Object {
                "process": [Function],
              }
            }
          />
          Host-supplier
        </li>
        <li
          className="Dropdown_main_body_element3"
          data-role="travel-administrator"
          onClick={[Function]}
        >
          <img
            alt="icon"
            src={
              Object {
                "process": [Function],
              }
            }
          />
          Travel Admin
        </li>
        <li
          className="Dropdown_main_body_element4"
          data-role="requester"
          onClick={[Function]}
        >
          <img
            alt="icon"
            src={
              Object {
                "process": [Function],
              }
            }
          />
          Requester
        </li>
      </ul>
    </div>
  </div>
</div>
`;
